### ISSUE TITLE
3.0.0-beta2 static libcrypto builds broken for armhf + armv8-a (e.g. tuned Pi3)

### ISSUE THREAD

**Issue by instinctify** [openssl/openssl#16273]
SYMPTOM: 'Exception in thread "main" java.lang.UnsatisfiedLinkError: undefined symbol: aes_v8_xts_encrypt' on LoadLibrary

APPARENT CAUSE
    openssl build tool seems to emits invalid symbol into static library when openssl built for armhf distro and armv8-a architecture

AFFECTS 
    openssl-3.0.0-beta2 (1.1.1h fine)

CONFIGURATION
```
ossl_opts="--release no-ssl3-method no-tls1-method no-tls1_1-method \
no-tls1_2-method no-dtls1-method no-dtls1_2-method no-tls no-dtls \
no-ssl no-aria no-bf no-blake2 no-camellia no-cast no-chacha no-des \
no-dsa no-idea no-md4 no-mdc2 no-poly1305 no-rc2 no-rc4 no-rmd160 \
no-siphash no-sm2 no-sm3 no-sm4 no-whirlpool no-afalgeng no-capieng \
no-cms no-comp no-ct no-dgram no-gost no-multiblock no-nextprotoneg \
no-ocsp no-rfc3779 no-srp no-srtp no-ts no-tests disable-unit-test \
disable-external-tests no-engine no-static-engine no-dynamic-engine \
no-hw-padlock no-sock no-autoload-config no-dso no-async no-psk \
no-rdrand no-posix-io no-ui-console no-sse2 no-ssl-trace no-legacy \
no-shared"

```
BOTH BUILDS
```
ach=armv8-a
osl=openssl-3.0.0-beta2
```

FAULTY BUILD (armhf + armv8-a)
```
pth=armhf-armv8-a
tgt=arm-linux-gnueabihf
srt=~/src/2020-12-02-raspios-buster-armhf # 32bit distro
osl_ach=linux-armv4
# TOOLCHAIN
cd ~/bld/$pth/binutils-2.31.1
~/src/binutils-2.31.1/configure --prefix=~/toolchains/$pth \
    --target=$tgt --with-arch=$ach --with-tune=cortex-a53
    --with-fpu=vfpv4 --with-float=hard
    --disable-multilib --with-sysroot=$srt
make; make install
cd ~/bld/$pth/gcc-8.3.0
~/src/gcc-8.3.0/configure --prefix=~/toolchains/$pth \
    --target=$tgt --with-arch=$ach --with-tune=cortex-a53 \
    --with-fpu=vfpv4 --with-float=hard \
    --disable-multilib --with-sysroot=$srt --enable-languages=c
make; make install
# OPENSSL BUILD
cd ~/src/$osl/$pth/build
CC=~/toolchains/$pth/bin/${tgt}-gcc
~/src/$osl/tar/Configure --prefix=~/src/$osl/$pth --libdir=lib \
    --openssldir=~/src/$osl/$pth ${osl_opts} ${osl_ach}
make build_libs; make install_dev
# OUTCOME
objdump --syms ~/src/$osl/$pth/lib/libcrypto.a | grep aes_v8_xts_encrypt
00000000         *UND*  00000000 aes_v8_xts_encrypt
00000000         *UND*  00000000 aes_v8_xts_encrypt
```

GOOD BUILD (aarch64 + armv8-a)
```
pth=aarch64-armv8-a
tgt=aarch64-linux-gnu
srt=~/src/2020-08-20-raspios-buster-arm64 # 64bit distro
osl_ach=linux-aarch64
# TOOLCHAIN
cd ~/bld/$pth/binutils-2.31.1
~/src/binutils-2.31.1/configure --prefix=~/toolchains/$pth \
    --target=$tgt --with-arch=$ach --with-cpu=cortex-a53 \
    --disable-multilib --with-sysroot=$srt
make; make install
cd ~/bld/$pth/gcc-8.3.0
~/src/gcc-8.3.0/configure --prefix=~/toolchains/$pth \
    --target=$tgt --with-arch=$ach--with-cpu=cortex-a53 \
    --disable-multilib --with-sysroot=$srt --enable-languages=c
make; make install
# OPENSSL BUILD
cd ~/src/$osl/$pth/build
CC=~/toolchains/$pth/bin/${tgt}-gcc
~/src/$osl/tar/Configure --prefix=~/src/$osl/$pth --libdir=lib \
    --openssldir=~/src/$osl/$pth ${osl_opts} ${osl_ach}
make build_libs; make install_dev
# OUTCOME
objdump --syms ~/src/$osl/$pth/lib/libcrypto.a | grep aes_v8_xts_encrypt
00000000000019a0 g     F .text  0000000000000894 aes_v8_xts_encrypt
0000000000000000         *UND*  0000000000000000 aes_v8_xts_encrypt
0000000000000000         *UND*  0000000000000000 aes_v8_xts_encrypt
```

**Comment by t8m** [comment]
Fix in #16275, please test.

**Comment by instinctify** [comment]
Made that exact change to line 95 of my vanilla ~/src/openssl-3.0.0-beta2/include/crypto/aes_platform.h

problem is solved. Thanks!

**Comment by instinctify** [comment]
Fixed by #16275

**Comment by t8m** [comment]
Reopening to let it get closed when the PR is merged.
